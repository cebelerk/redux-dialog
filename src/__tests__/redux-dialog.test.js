/* eslint-env jest */

import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import serializer from 'enzyme-to-json/serializer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'

import reduxDialog from '../redux-dialog';

const middlewares = []
const mockStore = configureStore(middlewares)

const BasicDialog = () => (
  <div>
    My awesome modalbox!
  </div>
)

BasicDialog.displayName = 'DialogComponent';

Enzyme.configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(serializer);

describe('reduxDialog', () => {
  it('should have the dialog open', () => {
    const initialState = {
      dialogReducer: {
        dialogs: {
          dialog1: {
            isOpen: true
          }
        }
      }
    }

    const Dialog = reduxDialog({
      name: 'dialog1'
    })(BasicDialog);

    const wrapper = mount(
      <Provider store={ mockStore(initialState) }>
        <Dialog />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should have the dialog closed', () => {
    const initialState = {
      dialogReducer: {
        dialogs: { }
      }
    }

    const Dialog = reduxDialog({
      name: 'dialog1'
    })(BasicDialog);

    const wrapper = mount(
      <Provider store={ mockStore(initialState) }>
        <Dialog />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('it should have the dialog open if isOpen is set to true', () => {
    const initialState = {
      dialogReducer: {
        dialogs: {
        }
      }
    }

    const Dialog = reduxDialog({
      name: 'dialog1',
      isOpen: true
    })(BasicDialog);

    const wrapper = mount(
      <Provider store={ mockStore(initialState) }>
        <Dialog />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('state takes presedence over the default isOpen props', () => {
    const initialState = {
      dialogReducer: {
        dialogs: {
          dialog1: {
            isOpen: false
          }
        }
      }
    }

    const Dialog = reduxDialog({
      name: 'dialog1',
      isOpen: true
    })(BasicDialog);

    const wrapper = mount(
      <Provider store={ mockStore(initialState) }>
        <Dialog />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});