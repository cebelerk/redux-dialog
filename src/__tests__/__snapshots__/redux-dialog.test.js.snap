// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduxDialog it should have the dialog open if isOpen is set to true 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ReduxDialog)>
    <ReduxDialog
      onRequestClose={[Function]}
      payload={Object {}}
    >
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="dialog1"
        isOpen={true}
        name="dialog1"
        onRequestClose={[Function]}
        parentSelector={[Function]}
        payload={Object {}}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="dialog1"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          name="dialog1"
          onRequestClose={[Function]}
          parentSelector={[Function]}
          payload={Object {}}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              aria-label="dialog1"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <DialogComponent
                onRequestClose={[Function]}
                payload={Object {}}
              >
                <div>
                  My awesome modalbox!
                </div>
              </DialogComponent>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </ReduxDialog>
  </Connect(ReduxDialog)>
</Provider>
`;

exports[`reduxDialog should have the dialog closed 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ReduxDialog)>
    <ReduxDialog
      onRequestClose={[Function]}
      payload={Object {}}
    >
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="dialog1"
        isOpen={false}
        name="dialog1"
        onRequestClose={[Function]}
        parentSelector={[Function]}
        payload={Object {}}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="dialog1"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={false}
          name="dialog1"
          onRequestClose={[Function]}
          parentSelector={[Function]}
          payload={Object {}}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        />
      </Modal>
    </ReduxDialog>
  </Connect(ReduxDialog)>
</Provider>
`;

exports[`reduxDialog should have the dialog open 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ReduxDialog)>
    <ReduxDialog
      isOpen={true}
      onRequestClose={[Function]}
    >
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="dialog1"
        isOpen={true}
        name="dialog1"
        onRequestClose={[Function]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="dialog1"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          name="dialog1"
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              aria-label="dialog1"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <DialogComponent
                isOpen={true}
                onRequestClose={[Function]}
              >
                <div>
                  My awesome modalbox!
                </div>
              </DialogComponent>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </ReduxDialog>
  </Connect(ReduxDialog)>
</Provider>
`;

exports[`reduxDialog state takes presedence over the default isOpen props 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ReduxDialog)>
    <ReduxDialog
      isOpen={false}
      onRequestClose={[Function]}
    >
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="dialog1"
        isOpen={false}
        name="dialog1"
        onRequestClose={[Function]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="dialog1"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={false}
          name="dialog1"
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        />
      </Modal>
    </ReduxDialog>
  </Connect(ReduxDialog)>
</Provider>
`;
